expression/builtin_time.go

```diff
@@ -7252,11 +7189,19 @@ func calAppropriateTime(minTime, maxTime, minSafeTime time.Time) time.Time {
 	return minSafeTime
 }
 
-func getFspByIntArg(ctx sessionctx.Context, exp Expression) (int, error) {
-	_, ok := exp.(*Constant)
+// getFspByIntArg is used by some time functions to get the result fsp. If len(expr) == 0, then the fsp is not explicit set, use 0 as default.
+func getFspByIntArg(ctx sessionctx.Context, exps []Expression) (int, error) {
```

Comment:
I think it is reasonable to change the argument `exps` from `[]Expression` to `Expression` to make it hard to misuse.

Reply:
I change it to []Expression to better reuse the same code. If you worrying about misusage, I think we can rename it or do something else.